hyperspy/signal.py

```diff
@@ -3867,6 +3867,157 @@ def add_gaussian_noise(self, std):
         self.events.data_changed.trigger(obj=self)
 
+    def transpose(self, signal_axes=None, navigation_axes=None, copy=False):
+        """Transposes the signal to have the required signal and navigation
+        axes.
+
+        Parameters
+        ----------
+        signal_axes, navigation_axes : {None, int, iterable}
+            With the exception of both parameters getting iterables, generally
+            one has to be None (i.e. "floating"). The other one specifies
+            either the required number or explicitly the axes to move to the
+            corresponding space.
+            If both are iterables, full control is given as long as all axes
+            are assigned to one space only.
+        copy : bool [False]
+            If the data should be re-ordered in memory, most likely making a
+            copy. Ensures the fastest available iteration at the expense of
+            memory.
+
+        See also
+        --------
+        transposed, T
+
+        Examples
+        --------
+        >>> s = hs.signals.BaseSignal(np.random.rand(1,2,3,4,5,6,7,8,9))
+        >>> s
+        <BaseSignal, title: , dimensions: (|9, 8, 7, 6, 5, 4, 3, 2, 1)>
+
+        >>> s.transpose() # just swap signal and navigation spaces
+        <BaseSignal, title: , dimensions: (9, 8, 7, 6, 5, 4, 3, 2, 1|)>
+
+        >>> s.transpose(signal_axes=5) # 5 axes in signal space
+        <BaseSignal, title: , dimensions: (9, 8, 7, 6|5, 4, 3, 2, 1)>
+
+        >>> s.transpose(navigation_axes=3) # 3 axes in navigation space
+        <BaseSignal, title: , dimensions: (9, 8, 7|6, 5, 4, 3, 2, 1)>
+
+        >>> # 3 explicitly defined axes in signal space
+        >>> s.transpose(signal_axes=[0, 2, 6])
+        <BaseSignal, title: , dimensions: (8, 6, 5, 4, 2, 1|9, 7, 3)>
+
+        >>> # A mix of two lists, but specifying all axes explicitly
+        >>> s.transpose(navigation_axes=[1, 2, 3, 4, 5, 8], signal_axes=[0, 6, 7])
+        <BaseSignal, title: , dimensions: (8, 7, 6, 5, 4, 1|9, 3, 2)>
+
+        """
+        from collections import Iterable
+
+        def iterable_not_string(thing):
+            return isinstance(thing, Iterable) and \
+                not isinstance(thing, str)
+
+        am = self.axes_manager
+        nat_axes = am._get_axes_in_natural_order()
+        if isinstance(signal_axes, int):
+            if navigation_axes is not None:
+                raise ValueError("The navigation_axes are not None, even "
+                                 "though just a number was given for "
+                                 "signal_axes")
+            if len(nat_axes) < signal_axes:
+                raise ValueError("Too many signal axes requested")
+            navigation_axes = nat_axes[:-signal_axes]
+            signal_axes = nat_axes[-signal_axes:]
+        elif iterable_not_string(signal_axes):
+            if navigation_axes is None:
+                signal_axes = tuple(am[ax] for ax in signal_axes)
```

Comment:
It would be cleared to move this line above `if navigation_axes is None:`  and remove line 3940.